docs/src/markdown/extensions/magiclink.md

```diff
@@ -479,6 +480,51 @@ and will copy them for your custom repository. If this is not sufficient, you ca
 `shortener_user_exclude` for your custom repository provider using your specified `name`. If you manually set excludes
 in this manner, no excludes from the same `type` will be copied over.
 
+## Simple User-Defined References
+
+You can integrate with additional services like ticketing systems or shortlink providers (sometimes called `go/` links) by defining simple patterns that will be autolinked.
+
+The `simplerefs` configuration supports matching a static prefix followed by an alphanumeric identifier. We look for patterns of form `{ref_prefix}{identifier}` with the following restrictions:
+
+- `ref_prefix` is a configured literal value consisting of a letter followed by alphanumeric characters or punctuation from the set `_/-`; it will usually end with punctuation like `TICKET-`
+- `identifier` is an alphanumeric string containing at least one character; it will be captured and injected into a configured target URL
+- The pattern must be preceeded by whitespace or be at the beginning of the line
+- The pattern must be followed by whitespace or punctuation from the set `.,!?:`
+
+Suppose we have a Jira project called `TICKET` where the issues are referred to like `TICKET-123` and a shortlink service where labels look like `go/myproject`. Our configuration will be:
```

Comment:
I'm not 100% sold on `go/` links. This is very tailored to a specific tracker. I'm not 100% sure if that will be too limited, not limited enough, I'm also a little worried about conflicts with some other plugins that may try and capture paths, or possibly future extension ideas. I probably need more time to think about such links.

Worst case, it can be considered in the future if I cannot come to a definitive answer now. Granted, I do understand that these would only capture refs that specifically match the prefix, not all paths with `/`, but worth noting I am not necessarily on board with this yet.

Reply:
I should further note, that I still don't even know exactly what a `go/` link is. It seems that it is not really an issue tracking link, but some specialty link? So it can have any identifier (not number tracker) after it? Is `go/` a specific well known and popular linking system? Are there others that mimic this system? I don't have enough knowledge on this to blindly accept this approach.

While it does appear that the URL uses `/` in it, do we have to use that as a separator for links? I suspect we do not. We could stick with our current notation and simply allow the user to choose the expectation of a number or id after the prefix: `go-<id>`. This could simply be done with some kind of `type` parameter: `"type": "num"` or `"type": "id"`.

Reply:
> I still don't even know exactly what a go/ link is

It's a pattern very much like a link shortening service like tinyurl, but internal to a company and generally people give meaningful names for the links rather than having them auto-generated.  There are some vendors like https://www.golinks.io/ that provide this as a service. And companies will sometimes provide DNS support or a browser plugin so that you can literally type `go/foo` into your browser and it will route it to `https://go.mycompany.com` and to the final destination.